app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -204,6 +206,10 @@ public void reportProgress(int progress) {
                 //Either way, don't re-do this step
                 this.publishProgress(PROGRESS_CLEANED);
 
+                if (isCancelled()) {
+                    // avoid making the http request if user cancelled the task
+                    return PullTaskResult.UNKNOWN_FAILURE;

```

Comment:
Yeah, results returned when cancelled get dropped. They don't need to be though, since later android APIs support passing values to `onCancelled`.
In this case I could return `null` or extend `PullTaskResult` to have a (unused) cancel result, both aren't perfect, but probably better than `UNKNOWN_FAILURE`. Follow-up suggestions?